### PR TITLE
Responsive demo app

### PR DESCRIPTION
**Pull request by factoidforrest**
Also updated material UI to 5 which made things look nicer, luckily they had an auto-upgrade tool.

### CODE REVIEW

**Comment by factoidforrest** [comment]
Derp gotta fix the unit tests

**Comment by factoidforrest** [comment]
Ok the first visit command fails, and then everything else works perfectly.  And only in CI.  Totally bizarre.  Either of you seen anything like this?

File: js/demo-apps/packages/react-front-end/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "test:url": "if [ -z ${APPLICATION_FRONT_END+x} ]; then echo 'http://localhost:3000'; else echo $APPLICATION_FRONT_END; fi",
-    "test:e2e": "cypress run --browser chrome $TEST_OPTIONS --config baseUrl=$(yarn run --silent test:url)",
+    "test:e2e": "cypress run --record --key 77270197-498a-4a68-b9ff-e5d6a0f04617 --browser chrome $TEST_OPTIONS --config baseUrl=$(yarn run --silent test:url)",
```
Comment (freeqaz) [inline js/demo-apps/packages/react-front-end/package.json:11]: this should be a secret value, and you should rotate this value (go delete it in your Cypress console). It's public now.

File: js/demo-apps/packages/react-front-end/src/common/components/Header.tsx [hunk 2]
```diff
@@ -46,22 +70,39 @@ export const Header: React.FunctionComponent<{
...
 
+  const sidebarOpen = useStoreState((state) => state.sidebarOpen);
+  const setSidebarOpen = useStoreActions((actions) => actions.setSidebarOpen);
+
   useEffect(() => {
     void loadUser({ transport: props.transport }); // Small hack to do this here but it makes sure the user is loaded whenever a page refreshes, ideally would happen in a dedicated component mounted in App
   }, [loadUser, props.transport]);
 
+  const theme = useTheme();
+  const onDesktop = useMediaQuery(theme.breakpoints.up('md'));
```
Comment (freeqaz) [inline js/demo-apps/packages/react-front-end/src/common/components/Header.tsx:81]: This is pretty cool to see.

File: js/demo-apps/packages/react-front-end/src/common/components/Header.tsx [hunk 3]
```diff
@@ -73,27 +114,40 @@ export const Header: React.FunctionComponent<{
...
     <AppBar position="fixed" className={classes.appBar}>
       <Toolbar>
-        <Typography variant="h6" noWrap className={classes.title}>
-          LunaSec Example App
-        </Typography>
-        <ToggleButtonGroup
-          className={classes.toggleButtonGroup}
-          exclusive
+        {/* hamburger icon shows the drawer on click */}
+        {!onDesktop ? (
```
Comment (freeqaz) [inline js/demo-apps/packages/react-front-end/src/common/components/Header.tsx:118]: nit: I prefer the syntax of something like this. It's flatter and easier to follow.
```tsx
function render() {
  const isDesktopMode = useMediaQuery(theme.breakpoints.up('md'));

  const mobileNavContent = (
    <IconButton
      className={classes.sideBarButton}
      edge="start"
      aria-label="open drawer"
      onClick={() => setSidebarOpen(!sidebarOpen)}
      size="large"
    >
      <MenuIcon />
    </IconButton>
  );
  
  const desktopNavContent = (
    <Typography variant="h6" noWrap className={classes.title}>
      LunaSec Example App
    </Typography>
  );

  const navContent = isDesktopMode ? desktopNavContent : mobileNavContent;
  
  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        {navContent}
        /* ... */
      </Toolbar>
    </AppBar>
  );
}
```

Alternatively, another style would be to just extract it into a functional component.

```tsx
function NavBar() {
  
  const isDesktopMode = useMediaQuery(theme.breakpoints.up('md'));
  
  if (isDesktopMode === false) {
    return (
      <IconButton
        className={classes.sideBarButton}
        edge="start"
        aria-label="open drawer"
        onClick={() => setSidebarOpen(!sidebarOpen)}
        size="large"
      >
        <MenuIcon />
      </IconButton>
    );
  }
  
  return (
    <Typography variant="h6" noWrap className={classes.title}>
      LunaSec Example App
    </Typography>
  );
}
```

File: js/demo-apps/packages/react-front-end/src/common/components/SideMenu.tsx [hunk 4]
```diff
@@ -122,20 +126,49 @@ export const SideMenu: React.FunctionComponent<{ mode: Mode }> = (props) => {
...
+          onClose={() => setSidebarOpen(false)}
+          classes={{
+            paper: classes.drawerPaper,
+          }}
+          onClick={() => setSidebarOpen(false)} // this should be automatic..but it's not
+        >
+          {slot}
+        </Drawer>
+      );
+    } else {
```
Comment (freeqaz) [inline js/demo-apps/packages/react-front-end/src/common/components/SideMenu.tsx:149]: Nit: Useless else can be removed here.

I should figure out how to add this to the linter. :thinking:

**Review by freeqaz — APPROVED** [review]
Overall looks good. Please rotate that Cypress key though!

**Comment by factoidforrest** [comment]
there's a possibility this CI issue is going to get fixed by the speed-up-ci changes that introduce static serving.  Wait for merge and try again

**Comment by factoidforrest** [comment]
merged these changes into the mega-merge that happened, they are in master

**Pull request closed without merging**